commit 6a9ba38e3ea9264cb041802cd7c067c3db49d3ed
Author: Laurent Montel <montel@kde.org>
Date:   Fri Mar 1 22:36:39 2019 +0100

    [knotification] compile without deprecated methods
    
    Summary: compile without deprecated method
    
    Test Plan: autotest ok
    
    Reviewers: dfaure
    
    Reviewed By: dfaure
    
    Subscribers: kde-frameworks-devel
    
    Tags: #frameworks
    
    Differential Revision: https://phabricator.kde.org/D19455

diff --git CMakeLists.txt CMakeLists.txt
index 7e9fd246a6b0867be6b9f7fbff4aaa3e2b72a600..509dff6c885801a56d90c1589c55294426a34b2b 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -94,7 +94,7 @@ if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ecm_install_po_files_as_qm(po)
 endif()
 if (NOT APPLE) # QtMac::setBadgeLabelText is deprecated
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
+add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050900)
 endif()
 add_subdirectory(src)
 if (BUILD_TESTING)
diff --git src/imageconverter.cpp src/imageconverter.cpp
index c98b6c5e90d57b59ca051dd80cf2aa6399449617..78a98254fba810fe126365161283ee5ddc8975a6 100644
--- src/imageconverter.cpp
+++ src/imageconverter.cpp
@@ -66,7 +66,11 @@ QVariant variantForImage(const QImage &_image)
 
 	QImage image = _image.convertToFormat(QImage::Format_RGBA8888);
 
-	QByteArray data((const char*)image.constBits(), image.sizeInBytes());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+    QByteArray data((const char*)image.constBits(), image.sizeInBytes());
+#else
+    QByteArray data((const char*)image.constBits(), image.byteCount());
+#endif
 
 	SpecImage specImage;
 	specImage.width = image.width();
diff --git src/kstatusnotifieritem.cpp src/kstatusnotifieritem.cpp
index d7467ea..1c4db31 100644
--- src/kstatusnotifieritem.cpp
+++ src/kstatusnotifieritem.cpp
@@ -1120,12 +1140,21 @@ KDbusImageStruct KStatusNotifierItemPrivate::imageToStruct(const QImage &image)
     KDbusImageStruct icon;
     icon.width = image.size().width();
     icon.height = image.size().height();
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     if (image.format() == QImage::Format_ARGB32) {
         icon.data = QByteArray((char *)image.bits(), image.sizeInBytes());
     } else {
         QImage image32 = image.convertToFormat(QImage::Format_ARGB32);
         icon.data = QByteArray((char *)image32.bits(), image32.sizeInBytes());
     }
+#else
+    if (image.format() == QImage::Format_ARGB32) {
+        icon.data = QByteArray((char *)image.bits(), image.byteCount());
+    } else {
+        QImage image32 = image.convertToFormat(QImage::Format_ARGB32);
+        icon.data = QByteArray((char *)image32.bits(), image32.byteCount());
+    }
+#endif
 
     //swap to network byte order if we are little endian
     if (QSysInfo::ByteOrder == QSysInfo::LittleEndian) {
